mesa/examples/basic/virus_on_network/model.py

```diff
@@ -60,23 +56,23 @@ def __init__(
             }
         )
 
-        # Create agents
-        for node in self.G.nodes():
-            a = VirusAgent(
-                self,
-                State.SUSCEPTIBLE,
-                self.virus_spread_chance,
-                self.virus_check_frequency,
-                self.recovery_chance,
-                self.gain_resistance_chance,
-            )
-
-            # Add the agent to the node
-            self.grid.place_agent(a, node)
+        VirusAgent.create_agents(
+            self,
+            num_nodes,
+            State.SUSCEPTIBLE,
+            virus_spread_chance,
+            virus_check_frequency,
+            recovery_chance,
+            gain_resistance_chance,
+            list(self.grid.all_cells),
+        )
 
         # Infect some nodes
-        infected_nodes = self.random.sample(list(self.G), self.initial_outbreak_size)
-        for a in self.grid.get_cell_list_contents(infected_nodes):
+        infected_nodes = CellCollection(
+            self.random.sample(list(self.grid.all_cells), self.initial_outbreak_size),
+            random=self.random,
+        )
```

Comment:
Non blocking, but might be interesting if we can make this API-wise a bit cleaner.

Reply:
Not sure what you are suggesting. I could write the example to not use a cell collection, but this kept the code closer to the original code and illustrates how users can create their own CellCollection.